File: lib/antlr-2.7.5/examples/java/heteroAST/BinaryOperatorAST.java
public abstract class BinaryOperatorAST extends CalcAST {
	public CalcAST left() {
		return (CalcAST)getFirstChild();
	}

	public CalcAST right() {
		CalcAST t = left();
		if ( t==null ) return null;
		return (CalcAST)t.getNextSibling();
	}
}

File: lib/antlr-2.7.5/examples/csharp/heteroAST/BinaryOperatorAST.cs
using System;

public abstract class BinaryOperatorAST : CalcAST {
	public CalcAST Left() {
		return (CalcAST)getFirstChild();
	}

	public CalcAST Right() {
		CalcAST t = Left();
		if ( t==null ) return null;
		return (CalcAST)t.getNextSibling();
	}
}
